src/Features/VisualBasic/Portable/CodeFixes/Spellcheck/SpellcheckCodeFixProvider.vb

```diff
@@ -36,190 +34,27 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeFixes.Spellcheck
-
-                ' Completion gave us the unbound identifier, so it's obviously okay. Bail.
-                If distance = 0 Then
-                    Return Nothing
-                End If
-
-                ' The old IDE code uses this as a metric of how well a string matches,
-                ' in addition to the edit distance and longest common substring length percentages.
-                Dim goodness = identifierText.Length - longestCommonSequence + distance
-
-                ' Calculate ED and LCS% against the longer of the two strings.
-                Dim maxLength = Math.Max(name.Length, identifierText.Length)
-
-                Dim editDistancePercentage = distance / maxLength
-                Dim longestCommonPercentage = longestCommonSequence / maxLength
-
-                ' If it's within tolerances, keep it.
-                If editDistancePercentage <= s_maximumEditDistancePercentage AndAlso longestCommonPercentage >= s_minimumLongestCommonSubsequencePercentage Then
-                    Dim textChange = completionRules.GetTextChange(item)
-                    results.Add(New SpellcheckResult(name, GetReasonableName(textChange.NewText), goodness))
-                End If
-            Next
-
-            If Not results.Any() Then
-                Return Nothing
-            End If
-
-            results.Sort(
-                Function(r1, r2)
-                    Dim goodnessComparisan = r1.Goodness.CompareTo(r2.Goodness)
-                    If goodnessComparisan <> 0 Then
-                        Return goodnessComparisan
-                    End If
-
-                    ' Try to order by name. Note that this is unescaped version of the name.
-                    Dim nameCompare = String.Compare(r1.Name, r2.Name, StringComparison.CurrentCultureIgnoreCase)
-                    If nameCompare <> 0 Then
-                        Return nameCompare
-                    End If
-
-                    ' Prefer escaped identifiers ahead of non-escaped
-                    Dim r1StartsWithEscape = r1.ReplacementText.Length > 0 AndAlso r1.ReplacementText(0) = "["c
-                    Dim r2StartsWithEscape = r2.ReplacementText.Length > 0 AndAlso r2.ReplacementText(0) = "["c
-
-                    If r1StartsWithEscape Then
-                        Return -1
-                    ElseIf r2StartsWithEscape
-                        Return 1
-                    End If
-
-                    Return String.Compare(r1.ReplacementText, r2.ReplacementText, StringComparison.CurrentCultureIgnoreCase)
-                End Function)
-
-            Return results _
-                .Select(Function(result) result.ReplacementText) _
-                .Distinct() _
-                .Take(s_maxMatches) _
-                .Select(Function(name) New SpellCheckCodeAction(document, identifierName, name))
+        Protected Overrides Function IsGeneric(completionItem As CompletionItem) As Boolean
+            Return completionItem.DisplayText.Contains("(Of")
```

Comment:
Thank you.  We can chat about ways we might be able to do this.  Maybe some sort of extension method so it's not on the core API (since TS can't implement it anyways).  Something like "TryGetOriginalSymbol" ?